### PR TITLE
[Task #1146] Reorganize recipes

### PR DESCRIPTION
**Pull request by vr4b4c**
## Summary

Previously, recipes needed to be carefully organised in a precise sequence, otherwise template wouldn't be successfully applied to the Rails application. The reason for careful organisation is because template is explicitly running `bundle install` at a certain point (`recipes/run_generators`), which meant changes to `Gemfile` had to be completed and should not be done afterwards (order coupling, and there is another problem with documenting this behaviour for the contributors). Completing bundle install marked the point where Rails generators could be run, but there was similar coupling as before, changes to Rails application configuration files had to be completed (`.env`, `config/database.yml`, ...). The idea was to run bundle install and rails generators as the last recipe (before committing the changes). But it didn't come without it's own problems, generators became disassociated from their natural recipes, e.g. flipper configuration is defined in the `recipes/flipper` but the generator and a piece of config which can only be added after the generator makes changes was defined in the `recipes/run_generators`. `recipes/run_generators` looks very chaotic at the moment.

The solution is to let Rails application generator run `bundle install` and hook into it's callback system using [after_bundle](https://guides.rubyonrails.org/rails_application_templates.html#after-bundle-block). One major coupling point is resolved as each recipe can specify it's own generators. There are still other coupling types present, `license_finder` recipe depends on `gemfile` recipe modifying the `Gemfile`, but it doesn't fall into scope of this task.

**Related issue**: [#1146](https://app.productive.io/1-infinum/projects/2274/tasks/task/13539880?board=268199&filter=MjA2MTY2)

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [x] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

File: recipes/license_finder.rb [hunk 1]
```diff
@@ -19,3 +19,15 @@
       - 'package.json'
   HEREDOC
 end
+
+after_bundle do
+  bundle_command 'exec license_finder permitted_licenses add "New BSD" "Simplified BSD" ruby "2-clause BSDL" ' \
+                 '"Python-2.0" MIT "Apache 2.0" ISC CC0-1.0 CC-BY-3.0 "Mozilla Public License 2.0" LGPL MIT-0 ' \
+                 '"Brakeman Public Use License" Apache "(CC-BY-4.0 AND OFL-1.1 AND MIT)" CC-BY-4.0 ' \
+                 '"MPL-2.0 OR Apache-2.0" "(MIT AND Zlib)" "(WTFPL OR MIT)" "PayPal SDK License" ' \
+                 '"MIT (http://mootools.net/license.txt)" "(MIT AND CC-BY-3.0)" 0BSD BSD '
```
Comment (lovro-bikic) [inline recipes/license_finder.rb:28]: I'd extract the licenses to an array for easier managing:
```ruby
bundle_command "exec license_finder permitted_licenses add #{permitted_licenses.join(' ')}"

**Review by djordjelacmanovic — APPROVED** [review]

File: recipes/license_finder.rb (hunk 1 above)
Comment (vr4b4c) [reply recipes/license_finder.rb:28]: Sounds reasonable, let's do it in a separate PR
